app/src/org/commcare/android/database/connect/models/ConnectJobRecord.java

```diff
@@ -461,12 +462,21 @@ public String getWorkingHours() {
         }
 
         try {
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss");
-            LocalTime start = LocalTime.parse(dailyStart, formatter);
-            LocalTime end = LocalTime.parse(dailyFinish, formatter);
-            formatter = DateTimeFormatter.ofPattern("h:mm a");
-
-            return formatter.format(start) + " - " + formatter.format(end);
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+                //DateTimeFormatter is more efficient than SimpleDateFormat
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss");
```

Comment:
nit: can we use a constant to refer to this pattern so that we only define this once in this class.